src/main/java/org/folio/edge/sip2/BootstrapVerticle.java

```diff
@@ -0,0 +1,58 @@
+package org.folio.edge.sip2;
+
+import io.vertx.config.ConfigRetriever;
+import io.vertx.config.ConfigRetrieverOptions;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonObject;
+import java.util.Objects;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class BootstrapVerticle extends AbstractVerticle {
+  private final Logger log = LogManager.getLogger();
+  private String deploymentID = null;
+  
+  @Override
+  public void start(Promise<Void> startFuture) {
+
+    log.info("Bootstrapping sip2, retrieving config from:\n {}", () -> config().encodePrettily());
+    JsonObject crOptionsJson = config().getJsonObject("tenantConfigRetrieverOptions");
+    ConfigRetrieverOptions crOptions = new ConfigRetrieverOptions(crOptionsJson);
+    ConfigRetriever configRetriever = ConfigRetriever.create(vertx, crOptions);
+
+    configRetriever.listen(change -> {
+      log.info("config change detected for sip2, redeploying MainVerticle");
+      deployMainVerticle(change.getNewConfiguration(), Promise.promise());
+    });    
+
+    configRetriever.getConfig(ar -> {
+      if (ar.succeeded()) {
+        log.info("Deploying sip2 MainVerticle");
+        deployMainVerticle(ar.result(), startFuture);
+      } else {
+        log.error("Unable to getConfig - {}", ar.cause().getMessage());
+        startFuture.fail(ar.cause());
+      }
+    });
+
+  }
+  
+  private void deployMainVerticle(JsonObject config, Promise<Void> completion) {
+    if (Objects.nonNull(deploymentID)) {
+      vertx.undeploy(deploymentID);
```

Comment:
Yes, I don't imagine config changes happen too often. We can look into setting the on-line status to 'N' if the kiosk cannot reconnect when connection is dropped on redeploy. It will take sub-seconds to redeploy, so hopefully the kiosk can gracefully recover.

Reply:
Still, I think we need to think about how we can avoid killing the connection to all tenants when it is likely that we are adding a new tenant and existing tenant config shouldn't be changed. I don't know what monitoring tenants do of their Kiosks' availability, but we don't want to draw attention to service stability, if possible.

Reply:
redeploy is no longer needed on config change

Reply:
redeploy has been removed